nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -256,6 +419,311 @@ impl DataStore {
             })
     }
 
+    pub async fn ip_pool_association_list(
+        &self,
+        opctx: &OpContext,
+        authz_pool: &authz::IpPool,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<IpPoolResource> {
+        use db::schema::ip_pool;
+        use db::schema::ip_pool_resource;
+
+        paginated(
+            ip_pool_resource::table,
+            ip_pool_resource::ip_pool_id,
+            pagparams,
+        )
+        .inner_join(ip_pool::table)
+        .filter(ip_pool::id.eq(authz_pool.id()))
+        .filter(ip_pool::time_deleted.is_null())
+        .select(IpPoolResource::as_select())
+        .load_async::<IpPoolResource>(
+            &*self.pool_connection_authorized(opctx).await?,
+        )
+        .await
+        .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+    }
+
+    // TODO: should this error on conflict instead of updating?
+    pub async fn ip_pool_associate_resource(
+        &self,
+        opctx: &OpContext,
+        ip_pool_resource: IpPoolResource,
+    ) -> CreateResult<IpPoolResource> {
+        use db::schema::ip_pool_resource::dsl;
+        opctx
+            .authorize(authz::Action::CreateChild, &authz::IP_POOL_LIST)
+            .await?;
```

Comment:
Perhaps unnecessary, but you could also check for modify on the silo? I assume mostly that if you have one you have the other but 🤷‍♂️

Reply:
Yeah, definitely. I need to change the signature to take the `authz_silo` and the `authz_pool`.